rpc/backend/blocks.go

```diff
@@ -296,28 +298,82 @@ func (b *Backend) EthMsgsFromTendermintBlock(
 		tx, err := b.clientCtx.TxConfig.TxDecoder()(tx)
 		if err != nil {
 			b.logger.Debug("failed to decode transaction in block", "height", block.Height, "error", err.Error())
+			// try to check if there is synthetic eth tx in tx result
+			ethMsg, additional := b.parseSyntheticTxFromBlockResults(txResults, i, tx, block)
+			if ethMsg != nil {
+				ethMsgs = append(ethMsgs, ethMsg)
+				txsAdditional = append(txsAdditional, additional)
+			}
```

Comment:
Wondering if it would make the code more readable if we abstract these four lines to a different function , as the same logic is also used in case error == nil , and msg != evmtypes.MsgEthereumTx)

Reply:
Also would it possible to call `parseSyntheticTxFromBlockResults` right in the beginning once , and add use the `ethMsg != nil` check , to append values and continue. 
And then check for `evmtypes.MsgEthereumTx ` .

Assuming ofcourse that `tx` cannot be both `MsgEthereumTx` and `SyntheticTX`